static/elements/chromedash-sample-panel.js

```diff
@@ -0,0 +1,178 @@
+
+    this.features = await (await fetch(url)).json();
+    this.features.forEach((feature) => {
+      feature.sample_links = feature.sample_links.map((link) => {
+        return link.replace(/github.com\/GoogleChrome\/samples\/tree\/gh-pages\/(.*)/i, 'googlechrome.github.io/samples/$1');
+      });
+    });
+
+    this.filtered = this.features;
+    this.filter(this.searchEl ? this.searchEl.value : '');
+    this._fireEvent('update-length', {length: this.filtered.length});
+  }
+
+  _computeIcon(el) {
+    return 'chromestatus:' + (el ? el.dataset.category : '');
+  }
+
+  _computeIconId(categoryStr) {
+    return 'chromestatus:' + this.categories[categoryStr];
+  }
+
+  _computeFeatureLink(id) {
+    return '/features/' + id;
+  }
+
+  _filterOnOperation(features, operator, version) {
+    return features.filter((feature) => {
+      const platformMilestones = [
+        parseInt(feature.shipped_milestone),
+        parseInt(feature.shipped_android_milestone),
+        parseInt(feature.shipped_ios_milestone),
+        parseInt(feature.shipped_webview_milestone),
+      ];
+      for (let i = 0, milestone; milestone = platformMilestones[i]; ++i) {
+        if (matchesMilestone_(milestone, operator, version)) {
+          return true; // Only one of the platforms needs to match.
+        }
+      }
+    });
+  }
+
+  // TODO: move filtering into a behavior. Mostly duped from chromedash-featurelist.html.
+  filter(query, category) {
+    let features = this.features;
+
+    if (!features) {
+      return;
+    }
+
+    if (!query && !category) {
+      if (history && history.replaceState) {
+        history.replaceState('', document.title, location.pathname + location.search);
+      } else {
+        location.hash = '';
+      }
+      this.filtered = features;
+      this._fireEvent('update-length', {length: this.filtered.length});
+
+      return;
+    }
+
+    if (query) {
+      if (history && history.replaceState) {
+        // TODO debounce this 500ms
+        history.replaceState({}, document.title, '/samples#' + query);
+      }
+
+      // Returns operator and version query e.g. ["<=25", "<=", "25"].
+      const operatorMatch = /^([<>=]=?)\s*([0-9]+)/.exec(query);
+      if (operatorMatch) {
+        features = this._filterOnOperation(features, operatorMatch[1], operatorMatch[2]);
+      } else {
+        const regex = new RegExp(query, 'i');
+        features = features.filter((feature) => {
+          return regex.test(feature.name) || regex.test(feature.summary);
+        });
+      }
+    }
+
+    // Further refine list based on selected category in menu.
+    if (category) {
+      const regex = new RegExp(category, 'i');
+      features = features.filter((feature) => {
+        return regex.test(feature.category);
+      });
+    }
+
+    this.filtered = features;
+
+    this._fireEvent('update-length', {length: this.filtered.length});
+  }
+
+  render() {
+    if (!this.filtered) {
+      return html``;
```

Comment:
you can return the `nothing` sentinel value from lit-html instead

Reply:
Good to know. Updated in all components.